[LordSaumya] iP

LordSaumya opened the pull request:
# Kronos
### The Lord of Time at the tips of your fingers 

> "Time management is for puny mortals like yourself. You will need my help."
> (Totally a real quote by [Kronos](http://https://en.wikipedia.org/wiki/Cronus "Kronos") himself) 

Kronos lets you manage your time like the Lord of Time himself :clock2:, featuring:
- A command-line-based interface
- Super easy commands
- Responses that will ~~annoy you~~ *keep you* on your toes.

#### Installation Instructions
1. Download the `.jar` file from latest release available [here](http://https://github.com/LordSaumya/ip/releases "here").
2. Open command prompt, navigate to the directory with the `.jar` file and enter the command `java -jar Duke.jar`.
3. ???
4. Profit

**Features:**
- [x] Easy-to-use commands
- [x] Sassy interface
- [ ] GUI (*coming soon*)

------------

For Java programmers, you can play around with the `run()` function too! Here it is:
```java
public void run() {
        ui.showStartMessage();

        Command command = new InvalidCommand();
        while (command.type != Command.CommandType.BYE) {
            command = parser.parseCommand(ui.readCommand());
            command.execute(list, ui);
        }

        // Write to file
		storage.writeFile(list);
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

LordSaumya committed:
Add .gitignore

LordSaumya committed:
Complete Level 0

LordSaumya committed:
Complete level 1

LordSaumya committed:
Complete Level 2

LordSaumya committed:
Ignore Task class files

LordSaumya committed:
Complete Level 3

LordSaumya committed:
Ignore class files

LordSaumya committed:
Complete Level 4

LordSaumya committed:
Add Text UI Testing

LordSaumya committed:
Complete Level 6

LordSaumya committed:
Complete Level 6

LordSaumya committed:
Test commit

LordSaumya committed:
Complete Level 7

LordSaumya committed:
Merge branch 'branch-Level-7'

LordSaumya committed:
Use LocalDateTime for datetimes

LordSaumya committed:
Fix implementation of DateTime to common format

LordSaumya committed:
Modify Duke to use LocalDateTime

LordSaumya committed:
Add exception handling for datetime parsing errors

LordSaumya committed:
Merge branch 'branch-Level-8'

LordSaumya committed:
Ignore UI class file & storage text  file

LordSaumya committed:
Add UI class to replace methods in the main class.

LordSaumya committed:
Ignore storage class files

LordSaumya committed:
Add and integrate Storage class with UI and Duke

LordSaumya committed:
Add command class and subclasses; Add parser class

LordSaumya committed:
Ignore class files

LordSaumya committed:
Add tasklist class & integrate with other classes

LordSaumya committed:
Separate main from run

LordSaumya committed:
Merge branch 'branch-A-MoreOOP'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

LordSaumya committed:
Ignore storage class file

LordSaumya committed:
Ignore vscode settings

LordSaumya committed:
Ignore class files

LordSaumya committed:
Add packages and change directory structure accordingly

LordSaumya committed:
Merge branch 'branch-A-Packages'

LordSaumya committed:
Add JavaDoc header comments for all classes and methods

LordSaumya committed:
Change names and indentation to match coding standards

LordSaumya committed:
Add find functionality

LordSaumya committed:
Merge branch 'branch-A-JavaDoc'

LordSaumya committed:
Merge branch 'branch-A-CodingStandard'

LordSaumya committed:
Merge branch 'branch-Level-9'

LordSaumya committed:
Add JavaDoc to FindCommand class

LordSaumya committed:
Ignore FindCommand class file

LordSaumya committed:
Merge remote-tracking branch 'origin/add-gradle-support'

LordSaumya committed:
Ignore task file

LordSaumya committed:
Add gradle build

LordSaumya committed:
Change gradle build file to use JUnit for testing

LordSaumya committed:
Override equality check method for commands

LordSaumya committed:
Add tests for Duke class

LordSaumya committed:
Override equality check methods for task classes

LordSaumya committed:
Add test cases for parser

LordSaumya committed:
Add test cases for storage class

LordSaumya committed:
Add test cases for task classes

LordSaumya committed:
Add test cases for UI class

LordSaumya committed:
Add test cases to command class

LordSaumya committed:
Configure Checkstyle

LordSaumya committed:
Add launcher for GUI

LordSaumya committed:
Add basic layout for GUI

LordSaumya committed:
Change commands to return strings as user inputs

LordSaumya committed:
Change Ui class to return strings for user inputs

LordSaumya committed:
Add profile pictures for Kronos and the user

LordSaumya committed:
Change Duke to use GUI

LordSaumya committed:
Add XML structures for DialgoueBox and MainWindow

LordSaumya committed:
Add controllers for the GUI components

LordSaumya committed:
Add Launcher and Main classes to implement GUI

LordSaumya committed:
Change Duke class to save to file on ByeCommand

LordSaumya committed:
Merge branch 'branch-Level-10'

LordSaumya committed:
Change parser and duke to use variable arguments

LordSaumya committed:
Merge branch 'branch-A-Varargs'

LordSaumya committed:
Fix bug in unmark command and store taskNum upon initialisation

LordSaumya committed:
Change parsertest inputs to String arrays for vararg compliance

brendanneojw commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,60 @@
+            try {
+                command.execute(list, ui);
+                storage.writeFile(list);
+            } catch (Exception e) {
```
Isn't this Pokemon Exception Handling that should be avoided?

brendanneojw commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,37 @@
+     */
+    public enum CommandType {
+        LIST, MARK, UNMARK, DELETE, TODO, DEADLINE, EVENT, BYE, FIND, INVALID
+    }
```
Nice usage of abstract class and enum to separate the different command types!

brendanneojw commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,119 @@
+    public Task parseFileTask(String fileTask) {
+        String[] taskComponents = fileTask.split(" \\| ");
+        String taskType = taskComponents[0];
+        boolean taskStatus = taskComponents[1].equals("1");
```
Since it is a boolean type, how about considering change the name to something like isMarked?

brendanneojw commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,90 @@
+                // Create new file
+                File taskFile = new File("tasks.txt");
+                taskFile.createNewFile();
+            } catch (Exception f) {
```
Maybe you can find out what the function createNewFile() throws as for the exception, rather than doing Pokemon Exception Handling.

brendanneojw reviewed (commented):
LGTM!

LordSaumya committed:
Add assertions to classes

There are several classes that require certain invariants to be true.
For example, the TaskList class should not have an uninitialised list.
We can use assertions to check that these invariants are true.

Assertions are used to check that certain conditions are true.

Let's add additional assertions as required to make sure all invariants are
satisfied when a method executes.

LordSaumya committed:
Merge pull request #1 from LordSaumya/A-Assertions

Complete A-Assertions

LordSaumya committed:
Add YML file for CI

LordSaumya committed:
Enable assertions in gradle build file

LordSaumya committed:
Remove unnecessary import from Duke

LordSaumya committed:
Change classes to use streams instead of loops

LordSaumya committed:
Refactor to remove magic literals and extract methods and make code clearer

LordSaumya committed:
Merge pull request #4 from LordSaumya/branch-A-Streams

Implement streams to replace loops

LordSaumya committed:
Merge branch 'master' into branch-A-CodeQuality

LordSaumya committed:
Merge pull request #3 from LordSaumya/branch-A-CodeQuality

Improve code quality

LordSaumya committed:
Add citations for images

LordSaumya committed:
Update README.md for final release

LordSaumya committed:
Add product screenshot
